src/components/Search/CompareWithBase.tsx

```diff
@@ -23,18 +26,31 @@ interface Expanded {
 
 function CompareWithBase({ mode }: CompareWithBaseProps) {
   const formWrapperRef = createRef<HTMLDivElement>();
+  const { enqueueSnackbar } = useSnackbar();
+  const { search } = useAppSelector((state) => state);
   const [base, setExpanded] = useState<Expanded>({
     expanded: true,
     class: 'expanded',
   });
-
   const [formHeight, setFormHeight] = useState<number>(400);
   const styles = CompareCardsStyles(mode, formHeight);
+  const baseRepository = search.base.repository;
+  const newRepository = search.new.repository;
+  const variant: VariantType = 'warning';
 
   useEffect(() => {
     setFormHeight(formWrapperRef.current?.clientHeight || formHeight);
   }, [formHeight]);
 
+  useEffect(() => {
+    if (
+      (baseRepository === 'try' && newRepository === 'mozilla-central') ||
+      (baseRepository === 'mozilla-central' && newRepository === 'try')
+    ) {
```

Comment:
It would be good to check with mozilla-beta and mozilla-release too. I've been told that the main issue that makes it difficult to compare is that there are different levels of optimization between the mozilla* branches and try (by default), but then there's no reason to single out mozilla-central.

Probably it would be good to explicit the no-warning pairs:
try, try
mozilla-XXX, mozilla-XXX

The check could be extractd to a function to make it easier to read here.

What do you think?

Reply:
I left a comment with some more context about this issue in the jira ticket: https://mozilla-hub.atlassian.net/browse/PCF-241

To summarize, I think we could simply display a warning anytime a try push is not being compared to another try push. For the other cases like m-c vs autoland, the differences in configurations aren't as much of an issue as try vs others.